repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -51,7 +54,9 @@ class Session(interpreter: Interpreter)
   private implicit val formats = DefaultFormats
 
   private var _state: SessionState = SessionState.NotStarted()
-  private var _history = IndexedSeq[Statement]()
+  private val _statements = mutable.Map[Int, Statement]()
```

Comment:
Java `LinkedHashMap` is another one for you to try. It's fine here, we could improve later.